main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -1019,4 +1035,10 @@ object Simplifier {
     visitExp(exp0)
   }
 
+  private def isPure(eff: Type): Purity = {
```

Comment:
Add comment!

Reply:
What happens if you do `if (eff == Type.Pure) Pure else if (..) else throw `? Does it throw?